pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix

```diff
@@ -46,22 +46,33 @@ in mkDerivation rec {
   postPatch = ''
     substituteInPlace Telegram/CMakeLists.txt \
       --replace '"''${TDESKTOP_LAUNCHER_BASENAME}.appdata.xml"' '"''${TDESKTOP_LAUNCHER_BASENAME}.metainfo.xml"'
+
+    substituteInPlace Telegram/ThirdParty/libtgvoip/os/linux/AudioInputALSA.cpp \
+      --replace '"libasound.so.2"' '"${alsa-lib}/lib/libasound.so.2"'
+    substituteInPlace Telegram/ThirdParty/libtgvoip/os/linux/AudioOutputALSA.cpp \
+      --replace '"libasound.so.2"' '"${alsa-lib}/lib/libasound.so.2"'
+    substituteInPlace Telegram/ThirdParty/libtgvoip/os/linux/AudioPulse.cpp \
+      --replace '"libpulse.so.0"' '"${libpulseaudio}/lib/libpulse.so.0"'
   '';
 
   # We want to run wrapProgram manually (with additional parameters)
   dontWrapGApps = true;
   dontWrapQtApps = true;
 
-  nativeBuildInputs = [ pkg-config cmake ninja python3 wrapGAppsHook wrapQtAppsHook removeReferencesTo ];
+  nativeBuildInputs = [
+    pkg-config cmake ninja python3 wrapGAppsHook wrapQtAppsHook
+    extra-cmake-modules
+  ];
 
   buildInputs = [
     qtbase qtimageformats gtk3 libsForQt5.kwayland libsForQt5.libdbusmenu lz4 xxHash
```

Comment:
```diff
-  qtbase qtimageformats gtk3 libsForQt5.kwayland libsForQt5.libdbusmenu lz4 xxHash
+  qtbase qtimageformats gtk3 kwayland libdbusmenu lz4 xxHash
```

`kwayland` and `libdbusmenu` are in the scope, the file is called using
`qt5.callPackage ../applications/networking/instant-messengers/telegram/tdesktop { }`

Reply:
Can you open a new PR to fix them? Does it causing any issues?

Reply:
It resulted in two places where Qt's version is specified, independently:

I backported new `tdesktop` to older NixOS, when default QT is `5.12`
Changing `qt5.callPackage` to `qt515.callPackage` was not enough to chose `5.15`, because these `libsForQt5` still pointed to default `pkgs.qt5` (which is 5.12).

As `tdesktop` tends to use bleeding edge versions of C++, Qt and other deps, eventually someone else will hit into the problem compiling it together with two different `Qt`s

Reply:
Doing a PR with the fixes.

Reply:
#132860